Description: Close bug  #827629.
             conflicting types for 'jpeg_mem_src' with
             libjpeg-turbo62 1.5.0
Author: Patrick Ohly <patrick.ohly@intel.com>
Reviewed-by: Herbert Parentes Fortes Neto <hpfn@ig.com.br>
Last-Update: 2016-06-19
Index: libgphoto2-2.5.10/camlibs/ax203/jpeg_memsrcdest.c
===================================================================
--- libgphoto2-2.5.10.orig/camlibs/ax203/jpeg_memsrcdest.c
+++ libgphoto2-2.5.10/camlibs/ax203/jpeg_memsrcdest.c
@@ -25,8 +25,8 @@
 #include "jpeg_memsrcdest.h"
 
 /* libjpeg8 and later come with their own (API compatible) memory source
-   and dest */
-#if JPEG_LIB_VERSION < 80
+   and dest, and older versions may have it backported */
+#if JPEG_LIB_VERSION < 80 && !defined(MEM_SRCDST_SUPPORTED)
 
 /* Expanded data source object for memory input */
 
Index: libgphoto2-2.5.10/camlibs/ax203/jpeg_memsrcdest.h
===================================================================
--- libgphoto2-2.5.10.orig/camlibs/ax203/jpeg_memsrcdest.h
+++ libgphoto2-2.5.10/camlibs/ax203/jpeg_memsrcdest.h
@@ -1,5 +1,7 @@
 #include <jpeglib.h>
 
+#if JPEG_LIB_VERSION < 80 && !defined(MEM_SRCDST_SUPPORTED)
+
 void
 jpeg_mem_src (j_decompress_ptr cinfo, unsigned char * buffer,
 	unsigned long bufsize);
@@ -7,3 +9,5 @@ jpeg_mem_src (j_decompress_ptr cinfo, un
 void
 jpeg_mem_dest (j_compress_ptr cinfo, unsigned char ** outbuffer,
 	unsigned long * outsize);
+
+#endif
Index: libgphoto2-2.5.10/camlibs/jl2005c/jpeg_memsrcdest.c
===================================================================
--- libgphoto2-2.5.10.orig/camlibs/jl2005c/jpeg_memsrcdest.c
+++ libgphoto2-2.5.10/camlibs/jl2005c/jpeg_memsrcdest.c
@@ -25,8 +25,8 @@
 #include "jpeg_memsrcdest.h"
 
 /* libjpeg8 and later come with their own (API compatible) memory source
-   and dest */
-#if JPEG_LIB_VERSION < 80
+   and dest, and older versions may have it backported */
+#if JPEG_LIB_VERSION < 80 && !defined(MEM_SRCDST_SUPPORTED)
 
 /* Expanded data source object for memory input */
 
Index: libgphoto2-2.5.10/camlibs/jl2005c/jpeg_memsrcdest.h
===================================================================
--- libgphoto2-2.5.10.orig/camlibs/jl2005c/jpeg_memsrcdest.h
+++ libgphoto2-2.5.10/camlibs/jl2005c/jpeg_memsrcdest.h
@@ -1,5 +1,7 @@
 #include <jpeglib.h>
 
+#if JPEG_LIB_VERSION < 80 && !defined(MEM_SRCDST_SUPPORTED)
+
 void
 jpeg_mem_src (j_decompress_ptr cinfo, unsigned char * buffer,
 	unsigned long bufsize);
@@ -7,3 +9,5 @@ jpeg_mem_src (j_decompress_ptr cinfo, un
 void
 jpeg_mem_dest (j_compress_ptr cinfo, unsigned char ** outbuffer,
 	unsigned long * outsize);
+
+#endif
